addons/website_forum/controllers/main.py

```diff
@@ -526,17 +538,16 @@ def post_flag(self, forum, post, **kwargs):
             return {'error': 'anonymous_user'}
         return post.flag()[0]
 
-    @http.route('/forum/<model("forum.forum"):forum>/post/<model("forum.post"):post>/ask_for_mark_as_offensive', type='http', auth="user", methods=['GET'], website=True)
-    def post_ask_for_mark_as_offensive(self, forum, post, **kwargs):
-        offensive_reasons = request.env['forum.post.reason'].search([('reason_type', '=', 'offensive')])
+    @http.route('/forum/<model("forum.post"):post>/ask_for_mark_as_offensive', type='json', auth="user", methods=['POST'], website=True)
+    def post_json_ask_for_mark_as_offensive(self, post, **kwargs):
+        if not post.can_moderate:
+            return {'error': _('%d karma required to mark a post as offensive.') % post.forum_id.karma_moderate}
+        values = self.ask_for_mark_as_offensive(post, **kwargs)
+        return {'mark_as_offensive' : request.env['ir.ui.view']._render_template('website_forum.mark_as_offensive', values)}
 
-        values = self._prepare_user_values(forum=forum)
-        values.update({
-            'question': post,
-            'forum': forum,
-            'reasons': offensive_reasons,
-            'offensive': True,
-        })
+    @http.route('/forum/<model("forum.forum"):forum>/post/<model("forum.post"):post>/ask_for_mark_as_offensive', type='http', auth="user", methods=['GET'], website=True)
+    def post_http_ask_for_mark_as_offensive(self, forum, post, **kwargs):
+        values = self.ask_for_mark_as_offensive(post, **kwargs)
```

Comment:
isn't this missing the post.can_moderate check?

Reply:
Done!